gdx-backend-jtransc/src/com/jtransc/media/limelibgdx/LimeInput.java

```diff
@@ -4,47 +4,135 @@
 import com.jtransc.annotation.JTranscMethodBody;
 import com.jtransc.annotation.haxe.HaxeMethodBody;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayDeque;
 
 public class LimeInput implements Input {
 
-	private static final int MAX_TOUCH_POINTS = 10;
+	private static final int MAX_TOUCH_POINTS = 20;
+
+	private static final int UNDEFINED = 0;
+	private static final int MOUSE_DOWN = 1;
+	private static final int MOUSE_UP = 2;
+	private static final int MOUSE_MOVE = 3;
+	private static final int MOUSE_WHEEL = 4;
+	private static final int TOUCH_START = 5;
+	private static final int TOUCH_END = 6;
+	private static final int TOUCH_MOVE = 7;
+	private static final int KEY_DOWN = 8;
+	private static final int KEY_TYPED = 9;
+	private static final int KEY_UP = 10;
+
+	@HaxeMethodBody("{% if extra.debugLimeInput %}return {{ extra.debugLimeInput }};{% else %}return false;{% end %}")
+	private static boolean isLimeInputDebug() {
+		return false;
+	}
 
 	private static boolean[] keys = new boolean[0x200];
 	private static boolean[] justPressed = new boolean[0x200];
 	private static boolean[] justReleased = new boolean[0x200];
-	private static boolean[] touchIndexes = new boolean[MAX_TOUCH_POINTS];
 
-	@HaxeMethodBody(
-		"{% if extra.debugLimeInput %}return {{ extra.debugLimeInput }};{% else %}return false;{% end %}"
-	)
-	private static boolean isLimeInputDebug() {
-		return false;
-	}
+	private static Pointer[] workingPointers = new Pointer[MAX_TOUCH_POINTS];
+	private static final ArrayDeque<Pointer> queuedPointers = new ArrayDeque<>();
+	private static final ArrayDeque<Pointer> freePointers = new ArrayDeque<>();
 
-	private static final HashMap<Integer, Pointer> pointers = new HashMap<>();
 	private static final Pointer mousePoint = new Pointer();
+
 	private static boolean lockMouse =
 		LimeDevice.getType() == Application.ApplicationType.iOS
 			|| LimeDevice.getType() == Application.ApplicationType.Android;
 
 	static private InputProcessor inputProcessor = new InputAdapter();
 
-	private static int getIndex() {
-		for (int i = 0; i < touchIndexes.length; i++) {
-			if (!touchIndexes[i]) {
-				touchIndexes[i] = true;
-				return i;
+	static {
+		for (int i = 0; i < MAX_TOUCH_POINTS; i++) {
+			workingPointers[i] = new Pointer();
+		}
+	}
+
+	static void flushInput() {
+		while (true) {
+			Pointer p;
+			synchronized (queuedPointers) {
```

Comment:
Is this called from multiple threads or why are we synchronizing?
Or did you copy this from gdx sources?